flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/catalog/CatalogTableITCase.scala

```diff
@@ -670,6 +670,409 @@ class CatalogTableITCase(isStreaming: Boolean) extends AbstractTestBase {
     assertEquals(expectedProperty, database.getProperties)
   }
+
+  @Test(expected = classOf[ValidationException])
+  def testCreateViewTwice(): Unit = {
+    val sourceData = List(
+      toRow(1, 1000),
+      toRow(2, 2000),
+      toRow(3, 3000)
+    )
+
+    TestCollectionTableFactory.initData(sourceData)
+
+    val sourceDDL =
+      """
+        |CREATE TABLE T1(
+        |  a int,
+        |  b int
+        |) with (
+        |  'connector' = 'COLLECTION'
+        |)
+      """.stripMargin
+
+    val sinkDDL =
+      """
+        |CREATE TABLE T2(
+        |  a int,
+        |  b int
+        |) with (
+        |  'connector' = 'COLLECTION'
+        |)
+      """.stripMargin
+
+    val viewDDL =
+      """
+        |CREATE VIEW T3(c, d) AS SELECT a, b FROM T1
+      """.stripMargin
+
+    tableEnv.sqlUpdate(sourceDDL)
+    tableEnv.sqlUpdate(sinkDDL)
+    tableEnv.sqlUpdate(viewDDL)
+    tableEnv.sqlUpdate(viewDDL)
+  }
+
+  @Test(expected = classOf[ValidationException])
+  def testCreateTemporaryViewTwice(): Unit = {
+    val sourceData = List(
+      toRow(1, 1000),
+      toRow(2, 2000),
+      toRow(3, 3000)
+    )
+
+    TestCollectionTableFactory.initData(sourceData)
+
+    val sourceDDL =
+      """
+        |CREATE TABLE T1(
+        |  a int,
+        |  b int
+        |) with (
+        |  'connector' = 'COLLECTION'
+        |)
+      """.stripMargin
+
+    val sinkDDL =
+      """
+        |CREATE TABLE T2(
+        |  a int,
+        |  b int
+        |) with (
+        |  'connector' = 'COLLECTION'
+        |)
+      """.stripMargin
+
+    val viewDDL =
+      """
+        |CREATE TEMPORARY VIEW T3(c, d) AS SELECT a, b FROM T1
+      """.stripMargin
+
+    tableEnv.sqlUpdate(sourceDDL)
+    tableEnv.sqlUpdate(sinkDDL)
+    tableEnv.sqlUpdate(viewDDL)
+    tableEnv.sqlUpdate(viewDDL)
+  }
+
+  @Test
+  def testCreateViewIfNotExistsTwice(): Unit = {
+    val sourceData = List(
+      toRow(1, 1000),
+      toRow(2, 2000),
+      toRow(3, 3000)
+    )
+
+    TestCollectionTableFactory.initData(sourceData)
+
+    val sourceDDL =
+      """
+        |CREATE TABLE T1(
+        |  a int,
+        |  b int
+        |) with (
+        |  'connector' = 'COLLECTION'
+        |)
+      """.stripMargin
+
+    val sinkDDL =
+      """
+        |CREATE TABLE T2(
+        |  a int,
+        |  b int
+        |) with (
+        |  'connector' = 'COLLECTION'
+        |)
+      """.stripMargin
+
+    val viewDDL =
+      """
+        |CREATE VIEW IF NOT EXISTS T3(c, d) AS SELECT a, b FROM T1
+      """.stripMargin
+
+    val query = "SELECT c, d FROM T3"
+
+    tableEnv.sqlUpdate(sourceDDL)
+    tableEnv.sqlUpdate(sinkDDL)
+    tableEnv.sqlUpdate(viewDDL)
+    tableEnv.sqlUpdate(viewDDL)
+
+    val result = tableEnv.sqlQuery(query)
+    result.insertInto("T2")
+    execJob("testJob")
+    assertEquals(sourceData.sorted, TestCollectionTableFactory.RESULT.sorted)
+  }
+
+  @Test
+  def testCreateTemporaryViewIfNotExistsTwice(): Unit = {
```

Comment:
add a test case like CREATE VIEW IF NOT EXISTS T3(d, e, f) AS SELECT * FROM T1

Reply:
Will add a case with `select *`